cpptod.dd

```diff
@@ -829,6 +830,84 @@ void test()
 }
 ------
 
+<hr>$(COMMENT  -------------------------------------------- )
+
+$(H3 <a name="interfaces">Interfaces</a>)
+
+Interfaces are used to describe the behavior or capabilities of a class without
+commiting to a particular implementation of the class.
+
+$(H4 The C++ Way)
+
+  In C++, an interface is implemented with an `Abstract Class`. The abstract class
+	defines the class capabilities as pure virtual functions. A class which implements
+  the interface inherits from the abstract class and implements the pure virtual functions.
+
+$(CPPCODE
+class Interface {
+  public:
+    virtual int function() = 0;
+};
+
+class FirstVariantOfImplementation : public Interface {
+  public:
+    int function() {
+      ...
+    }
+}
+
+class SecondVariantOfImplementation : public Interface {
+  public:
+    int function() {
+      ...
+    }
+}
+)
+
+The interface is used as follows:
+
+$(CPPCODE
+FirstVariantOfImplementation FirstVariant;
+SecondVariantOfImplementation SecondVariant;
+
+FirstVariant.function();
+SecondVariant.function();
+)
+
+$(H4 The D Way)
+
+  In D, interfaces (Alexandrescu 2010, p. 212; Cehreli 2017, p. 347) are implemented
+	using the `interface` keyword. The implementation class inherits from the interface
+	and implements the interface functions.
```

Comment:
In D, interfaces (Alexandrescu 2010, p. 212; Cehreli 2017, p. 347) are implemented with the help of the `interface` keyword, which introduces a class containing mainly unimplemented method declarations (non-static method definitions and non-static data members are not allowed). The implementation class inherits one or more interfaces and implements the interface methods."